orttraining/orttraining/test/training_api/core/checkpoint_test.cc

```diff
@@ -160,6 +160,35 @@ TEST(CheckpointApiTest, SaveOnnxModelAsCheckpoint_ThenLoad_CPU) {
   }
 }
 
+/**
+ * Load ONNX model with parameters set to 0 from file path, Load Checkpoint weights into the Model,
+ * Then compare the new weights to 0 to make sure they were changed after loading checkpoint to model.
+ */
+TEST(CheckpointApiTest, LoadCheckpointToModel) {
+  // Phase 1: Load a Model with weights set to zero.
+  auto model_uri = MODEL_FOLDER "transform/load_checkpoint/zero_model.onnx";
+  ONNX_NAMESPACE::ModelProto p_model;
+  ORT_ENFORCE(Model::Load(model_uri, p_model).IsOK());
+
+  // Phase 2: Load the checkpoint weights into the Model.
+  // Call Load APIs
+  PathString checkpoint_path = ORT_TSTR("testdata/transform/load_checkpoint");
+  ASSERT_STATUS_OK(LoadCheckpointToModel(checkpoint_path, p_model));
+
+  // Phase 3: Make sure the Model's weights are not equal to zero after loading the new ones.
+  // Load imported initializers into the Model
+  for (auto& init : *(p_model.mutable_graph()->mutable_initializer())) {
+    // Convert the tensor bytes to a float array to compare.
+    size_t len = init.raw_data().size() / sizeof(float);
+    float float_values[len];
```

Comment:
This is probably not legal C++ given you're trying to assign variable length array on the stack.
Could you try to find an alternative to this approach that is more `C++`y?